widget/select.go

```diff
@@ -107,20 +110,67 @@ type Select struct {
 	PlaceHolder string
 	OnChanged   func(string) `json:"-"`
 
-	hovered, tapped bool
-	popUp           *PopUpMenu
+	hovered bool
+	focused bool
+	tapped  bool
+	popUp   *PopUpMenu
+	Parent  fyne.CanvasObject
 }
 
 var _ fyne.Widget = (*Select)(nil)
 
+// FocusGained is called when the Entry has been given focus.
+func (s *Select) FocusGained() {
+	s.focused = true
+	s.Refresh()
+}
+
+// FocusLost is called when the Entry has had focus removed.
+func (s *Select) FocusLost() {
+	s.focused = false
+	s.Refresh()
+}
+
+// Focused returns whether or not this Entry has focus.
+func (s *Select) Focused() bool {
+	return s.focused
+}
+
+func (s *Select) TypedRune(r rune) {
+}
+
+func (s *Select) TypedKey(key *fyne.KeyEvent) {
+	if key.Name == fyne.KeyEnter || key.Name == fyne.KeyReturn || key.Name == fyne.KeySpace {
+		if s.popUp==nil || !s.popUp.Visible() {
+			s.Tapped(nil)
+		} else {
+			s.popUp.Menu.Do()
+		}
+	}
+	if s.popUp!=nil {
```

Comment:
With your overlay fix in place this code will not execute as the popup itself will get events?...

Reply:
Yes, except for the Enter keys. The code is now simplified.